include/engine/datafacade/contiguous_internalmem_datafacade.hpp

```diff
@@ -856,19 +859,19 @@ class ContiguousInternalMemoryDataFacade : public BaseDataFacade
         return result_datasources;
     }
 
-    virtual std::string GetDatasourceName(const uint8_t datasource_name_id) const override final
+    virtual StringView GetDatasourceName(const DatasourceID id) const override final
     {
         BOOST_ASSERT(m_datasource_name_offsets.size() >= 1);
-        BOOST_ASSERT(m_datasource_name_offsets.size() > datasource_name_id);
+        BOOST_ASSERT(m_datasource_name_offsets.size() > id);
 
-        std::string result;
-        result.reserve(m_datasource_name_lengths[datasource_name_id]);
-        std::copy(m_datasource_name_data.begin() + m_datasource_name_offsets[datasource_name_id],
-                  m_datasource_name_data.begin() + m_datasource_name_offsets[datasource_name_id] +
-                      m_datasource_name_lengths[datasource_name_id],
-                  std::back_inserter(result));
+        auto first = m_datasource_name_data.begin() + m_datasource_name_offsets[id];
+        auto last = m_datasource_name_data.begin() + m_datasource_name_offsets[id] +
+                    m_datasource_name_lengths[id];
+        // These iterators are useless: they're InputIterators onto a contiguous block of memory.
+        // Deref to get to the first element, then Addressof to get the memory address of the it.
+        const std::size_t len = &*last - &*first;
```

Comment:
Because these are pointers. Always.
Also `operator-` is guaranteed to work for random access iterators anyway.